### PR TITLE
Support Python 3.12

### LINKED ISSUE #5680: [Feature request] Python 3.12 support
Support Python 3.12

### PR DESCRIPTION
**Pull request by isdanni**
### Description
Close #5680

### Motivation and Context
- Add Python 3.12 support.

### CODE REVIEW

**Comment by justinchuby** [comment]
Thank you so much! Could you also enable one for the CI pipeline: https://github.com/onnx/onnx/blob/2cf3749d475e513cb9a2124f89308e214d4020db/.azure-pipelines/Linux-CI.yml#L40. 

something like this should do

```
      py312-internal-protobuf:
        python.version: '3.12'
        onnx_ml: 1
        build_type: 'Release'
        documentation: 0
        protobuf_type: 'Internal'
```

**Comment by justinchuby** [comment]
Looks like protobuf 3.12 supported was merged just a few hours ago :) We may need to wait on this

File: .github/workflows/release_linux_aarch64.yml [hunk 1]
```diff
@@ -98,38 +98,14 @@ jobs:
...
-          ${{ env.img }} \
-          bash -exc '\
-          source .env/bin/activate && \
-          python -m pip uninstall -y onnx numpy && python -m pip install numpy==1.23.2 && \
-          python -m pip install dist/*manylinux2014_aarch64.whl && \
-          pytest && \
-          deactivate'
-        fi
-
     - name: Verify ONNX with the minimum supported protobuf (from requirements.txt)
```
Comment (justinchuby) [inline .github/workflows/release_linux_aarch64.yml:101]:
```suggestion
    - name: Verify ONNX with the minimumly supported packages
```

File: .github/workflows/release_win.yml [hunk 2]
```diff
@@ -121,11 +108,11 @@ jobs:
         Get-ChildItem -Path dist/*.whl | foreach {python -m pip install --upgrade $_.fullname}
         pytest
 
-    - name: Verify ONNX with the minimum supported protobuf (from requirements.txt)
+    - name: Verify ONNX with the minimumly supported packages
       if: ${{ always() }}
       run: |
         cd onnx
-        python -m pip uninstall -y protobuf onnx && python -m pip install protobuf==3.20.2
+        python -m pip uninstall -y protobuf numpy onnx && python -m pip install -r requirements.txt
```
Comment (justinchuby) [inline .github/workflows/release_win.yml:115]:
```suggestion
        python -m pip uninstall -y protobuf numpy onnx && python -m pip install -r requirements-min.txt
```

**Comment by justinchuby** [comment]
Looks like pypa's docker file doesn't have python3.12 install in the expected path. Not sure why yet

**Comment by isdanni** [comment]
> Looks like pypa's docker file doesn't have python3.12 install in the expected path. Not sure why yet

Marking this as draft at the moment so it won't fail weekly CI. Will reopen once the package issue is resolved.

**Comment by justinchuby** [comment]
TODO: Fix the skip in https://github.com/onnx/onnx/pull/5743/files#diff-ecc29c1ce677ebf633712a033d0e1989082afd3a054d304aec3fbe3f6252cd38R93

**Review by justinchuby — APPROVED** [review]

**Pull request merged**